web_widget_text_count/static/src/js/fields_text_count.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('text_count', function(require) {
+    "use strict";
+
+    require('web.dom_ready');
+    var registry = require('web.field_registry');
+    var basicFields = require('web.basic_fields');
+    var FieldText = basicFields.FieldText;
+
+    var FieldTextCount = FieldText.extend({
+        events: _.extend({}, FieldText.prototype.events, {
+            'input': 'count_char',
+        }),
+        supportedFieldTypes: ['text'],
+
+        start: function () {
+            
+            var self = this;
+            return this._super().then(function(){
+                if (self.mode === 'edit') {
+                    if (self.attrs.size) {
+                        // this place can be improved should relay on self.field.size
```

Comment:
this comment can be removed. In any case is good to support size on demand per view.